docs/superbench-config.mdx

```diff
@@ -0,0 +1,389 @@
+---
+id: superbench-config
+---
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+
+# SuperBench Config File
+
+[YAML](https://yaml.org/spec/1.2/spec.html) format configuration file is an efficient method to take full advantage of SuperBench.
+You can put it in any place and specify the path to config file through `-c /path/to/config.yaml` in `sb` CLI.
+
+This document covers schema of SuperBench configuration YAML file.
+To learn the basics of YAML, see [Learn YAML in Y minutes](https://learnxinyminutes.com/docs/yaml/).
+SuperBench configuration supports most of the YAML features, including anchors and aliases, merge key, etc.
+
+## Conventions
+
+Here lists syntax convertions used in this document:
+
+* The schema and example are in YAML format.
+
+* In YAML mappings which use a colon `:` to mark `key: value` pair.
+
+  The left side of colon is a literal keyword defined in configuration,
+  if it is surrounded by `${}`, like `${name}`, then the key is a string that can be defined by user.
+
+  The right side of colon is a data type, which may be a Python built-in types (like `string`, `dict`),
+  or a rich structure defined in this document (first character capitalized).
+
+* The notation `[ datatype ]` indicates a YAML sequence of the mentioned data type.
+  For example, `[ string ]` is a list of strings.
+
+* The notation `|` indicates there are multiple optional data types.
+  For example, `string | [ string ]` means either a string or a list of strings is allowed.
+
+## Configuration Schema
+
+The configuration file describes all benchmarks running by SuperBench.
+There will be one or more benchmarks, each benchmark has its own settings and parameters.
+One benchmark may have one or more modes, which indicates how to run benchmarks in all given machines.
+
+Here is an overview of SuperBench configuration structure:
+
+<Tabs
+  defaultValue='schema'
+  values={[
+    {label: 'Schema', value: 'schema'},
+    {label: 'Example', value: 'example'},
+  ]
+}>
+<TabItem value='schema'>
+
+```yaml
+version: string
+superbench:
+  enable: string | [ string ]
+  var:
+    ${var_name}: dict
+  benchmarks:
+    ${benchmark_name}: Benchmark
+```
+
+</TabItem>
+<TabItem value='example'>
+
+```yaml
+version: v0.2
+superbench:
+  enable: benchmark_1
+  var:
+    var_1: value
+  benchmarks:
+    benchmark_1:
+      enable: true
+      modes:
+      - name: local
+```
+
+</TabItem>
+</Tabs>
+
+### `version`
+
+Version of the configuration file.
```

Comment:
what happens if one uses incorrect version number, something like 2.0 instead of 0.2?

Reply:
currently there's no validation check for this field, it's reserved for backward capability in the future.
will add validations for the config later and may throw an error if version is mismatched.